perllib/FixMyStreet/App/Controller/Root.pm

```diff
@@ -130,6 +142,27 @@ sub page_error : Private {
     $c->response->status($code);
 }
 
+sub check_login_required : Private {
+    my ($self, $c) = @_;
+
+    return if $c->user_exists || !FixMyStreet->config('LOGIN_REQUIRED');
+
+    my $whitelist = [
+        '^auth$',
+        '^auth/',
+        '^js/translation_strings\.[a-zA-Z]{2}(-[a-zA-Z]{2})?\.js',
+        '^[PACQM]{1}/', # various tokens that log the user in
+    ];
+
+    foreach (@$whitelist) {
```

Comment:
No reason for this to be an array reference, plus you could join the entries with `|` and do one regex? In fact, with `x` you could write it out as a multiline regex, think that's as clear, e.g.
```
my $whitelist = qr{
    ^auth(/|$)
  | ^js/translation_strings\.(.*?)\.js
  | ^[PACQM]/  # various tokens that log the user in
}x;
return if $c->request->path =~ $whitelist;
```